modules/tinymce/src/core/main/ts/content/ContentTypes.ts

```diff
@@ -10,23 +10,42 @@ import AstNode from '../api/html/Node';
 export type Content = string | AstNode;
 export type ContentFormat = 'raw' | 'text' | 'html' | 'tree';
 
-export interface GetContentArgs {
-  format?: ContentFormat;
-  get?: boolean;
-  content?: string;
-  getInner?: boolean;
+interface GetContentArgsBase {
+  format: ContentFormat;
+  get: boolean;
+  getInner: boolean;
   no_events?: boolean;
   [key: string]: any;
 }
 
-export interface SetContentArgs {
-  format?: string;
-  set?: boolean;
-  content?: string;
+interface SetContentArgsBase {
+  format: string;
+  set: boolean;
+  content: Content;
   no_events?: boolean;
   no_selection?: boolean;
 }
 
+export interface GetContentArgs extends GetContentArgsBase {
+  content: Content;
```

Comment:
Are we sure this is correct? Looking at the `GetContentImpl` logic, it'll only ever use a string in the event. Maybe we do need to break this up into 2 types, one for the event and one for the args? Additionally, the more I look this it looks like `content` should be optional here as it's not added into **after** `BeforeGetContent` fires, so we may even need 3 types (2 for the different events and 1 for the args).

- [ ] Ensure this is correct, as per the above comments

Reply:
Actually the same is true of `SetContentImpl` as well, in that case `content` will always be a `string` since it serializes the `AstNode`. So I definitely think we're going to need 2 or more different types here (which will likely extend a base type).

Reply:
I'll continue to use the single type for this PR, but I'll add a note on the separate issue to properly fix this that this should be done.